pkg/controller/operatingsystemconfig/os-common/generator/generator.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package generator
+
+// Generator renders an OperatingSystemConfig into a
+// representation suitable for an specific OS
+type Generator interface {
+	Generate(*OperatingSystemConfig) ([]byte, *string, error)
```

Comment:
As of now, the result type `([]byte, *string, error)` is not really descriptive - maybe add named result types?
Also, I think that the `Generate` call should not return the `cmd *string` as this is anyways a 'constant'. The `cmd` should be anyways available for the caller.

Reply:
The command is fixed for an operating system controller, but not to the library, I just moved the logic from the actuator to the generator, so that each os can implement a different logic.

Reply:
+1 with named result. types.